### PR TITLE
UserPreference Service Gateway

### PR DESCRIPTION
**Pull request by nmonterroso**
@Wikia/services-team @pchojnacki 
https://wikia-inc.atlassian.net/browse/SERVICES-984

Creating a UserPreference gateway using `swaggerGen project(":service:user-preference")`. In doing this I noticed that the annotation packages are different with what the resource uses (`com.wordnik.swagger.annotations` vs `io.swagger.annotations`) because the generation uses the latest version of Swagger, but the user preference annotations are using the old version of Swagger. To fix this, I updated our `dropwizard-swagger` integration to use the same version of Swagger as the code generation, since it's a more-current version. Because of this, I had to update all the resources to import the correct packages. 

@pchojnacki - with this upgrade, how do you feel about removing `:lib:io-swagger` and re-consolidating on `:lib:swagger` since it's now up to date with the Swagger annotations? iirc the reason for a new project was because the old project wasn't up to date.

### CODE REVIEW

File: service/discussion/src/main/java/com/wikia/discussionservice/resources/ForumResource.java [hunk 1]
```diff
@@ -140,7 +138,7 @@ public Response getForums(
       @ApiResponse(code = 404, message = "No forum found", response = HalProblem.class)})
   @Timed
   public Response getForum(
```
Comment (WikiaTech) [inline service/discussion/src/main/java/com/wikia/discussionservice/resources/ForumResource.java]: MAJOR Method has 11 parameters, which is greater than 7 authorized. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00107)

File: service/discussion/src/main/java/com/wikia/discussionservice/resources/ForumResource.java [hunk 2]
```diff
@@ -222,7 +220,7 @@ public Response getForum(
       @ApiResponse(code = 404, message = "No forum found", response = HalProblem.class)})
   @Timed
   public Response getNewer(
```
Comment (WikiaTech) [inline service/discussion/src/main/java/com/wikia/discussionservice/resources/ForumResource.java]: MAJOR Method has 8 parameters, which is greater than 7 authorized. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00107)

File: service/discussion/src/main/java/com/wikia/discussionservice/resources/PostResource.java [hunk 3]
```diff
@@ -250,7 +248,7 @@ public Response lockPost(
   @RequiresPermissions("posts:lock") // Same permission as locking
```
Comment (WikiaTech) [inline service/discussion/src/main/java/com/wikia/discussionservice/resources/PostResource.java]: MINOR Move this trailing comment on the previous empty line. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ATrailingCommentCheck)

File: service/discussion/src/main/java/com/wikia/discussionservice/resources/ThreadResource.java [hunk 4]
```diff
@@ -116,7 +114,7 @@ public ThreadResource(ThreadService threadService,
       @ApiResponse(code = 404, message = "No thread found", response = HalProblem.class)})
   @Timed
   public Response getThread(
```
Comment (WikiaTech) [inline service/discussion/src/main/java/com/wikia/discussionservice/resources/ThreadResource.java]: MAJOR Method has 11 parameters, which is greater than 7 authorized. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00107)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 18 issues:
* MAJOR 9 major
* MINOR 4 minor
* INFO 5 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Return an empty collection instead of null.](https://github.com/Wikia/pandora/blob/6ba50fed4452e1fe7d023bd6e3f6265f62b8d36c/service/discussion/src/main/java/com/wikia/discussionservice/domain/hal/output/ForumThreadHalResponse.java#L40) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1168)
* MAJOR [This block of commented-out lines of code should be removed.](https://github.com/Wikia/pandora/blob/6ba50fed4452e1fe7d023bd6e3f6265f62b8d36c/service/discussion/src/main/java/com/wikia/discussionservice/domain/hal/output/PostHalResponse.java#L42) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
* MAJOR [Return an empty collection instead of null.](https://github.com/Wikia/pandora/blob/6ba50fed4452e1fe7d023bd6e3f6265f62b8d36c/service/discussion/src/main/java/com/wikia/discussionservice/domain/hal/output/PostHalResponse.java#L48) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1168)
* MINOR [Define a constant instead of duplicating this literal "No forum found for site id: %s with forum id: %s" 4 times.](https://github.com/Wikia/pandora/blob/6ba50fed4452e1fe7d023bd6e3f6265f62b8d36c/service/discussion/src/main/java/com/wikia/discussionservice/resources/ForumResource.java#L209) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "Invalid response group" 3 times.](https://github.com/Wikia/pandora/blob/6ba50fed4452e1fe7d023bd6e3f6265f62b8d36c/service/discussion/src/main/java/com/wikia/discussionservice/resources/ForumResource.java#L114) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* INFO [Complete the task associated to this TODO comment.](https://github.com/Wikia/pandora/blob/6ba50fed4452e1fe7d023bd6e3f6265f62b8d36c/service/discussion/src/main/java/com/wikia/discussionservice/resources/PostResource.java#L78) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)
* INFO [Complete the task associated to this TODO comment.](https://github.com/Wikia/pandora/blob/6ba50fed4452e1fe7d023bd6e3f6265f62b8d36c/service/discussion/src/main/java/com/wikia/discussionservice/resources/PostResource.java#L112) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)
* INFO [Complete the task associated to this TODO comment.](https://github.com/Wikia/pandora/blob/6ba50fed4452e1fe7d023bd6e3f6265f62b8d36c/service/discussion/src/main/java/com/wikia/discussionservice/resources/PostResource.java#L283) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)
* MAJOR [Either log or rethrow this exception.](https://github.com/Wikia/pandora/blob/6ba50fed4452e1fe7d023bd6e3f6265f62b8d36c/service/discussion/src/main/java/com/wikia/discussionservice/resources/PostResource.java#L228) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)
* MAJOR [Either log or rethrow this exception.](https://github.com/Wikia/pandora/blob/6ba50fed4452e1fe7d023bd6e3f6265f62b8d36c/service/discussion/src/main/java/com/wikia/discussionservice/resources/PostResource.java#L260) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)
* INFO [Complete the task associated to this TODO comment.](https://github.com/Wikia/pandora/blob/6ba50fed4452e1fe7d023bd6e3f6265f62b8d36c/service/discussion/src/main/java/com/wikia/discussionservice/resources/ThreadResource.java#L94) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)
* INFO [Complete the task associated to this TODO comment.](https://github.com/Wikia/pandora/blob/6ba50fed4452e1fe7d023bd6e3f6265f62b8d36c/service/discussion/src/main/java/com/wikia/discussionservice/resources/ThreadResource.java#L285) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)
* MINOR [Move this constructor to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/6ba50fed4452e1fe7d023bd6e3f6265f62b8d36c/service/user-attribute/src/main/java/com/wikia/services/userattribute/resources/AttributeResource.java#L48) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MAJOR [Remove those useless parentheses.](https://github.com/Wikia/pandora/blob/6ba50fed4452e1fe7d023bd6e3f6265f62b8d36c/service/user-preference/src/main/java/com/wikia/userpreference/UserPreferenceApplication.java#L35) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessParenthesesCheck)

**Comment by pchojnacki** [comment]
>@pchojnacki - with this upgrade, how do you feel about removing :lib:io-swagger and re-consolidating on :lib:swagger since it's now up to date with the Swagger annotations? iirc the reason for a new project was because the old project wasn't up to date.
:+1: 
I only added io-swagger to avoid forcing everyone to update to new version. So if you consolidate the two it would be awesome.

**Comment by pchojnacki** [comment]
Since checks pass then LGTM

**Comment by nmonterroso** [comment]
Cool, I've created https://wikia-inc.atlassian.net/browse/SERVICES-1006 to track the consolidation on `:lib:swagger`

**Comment by WikiaTech** [comment]
SonarQube analysis reported 18 issues:
* MAJOR 9 major
* MINOR 4 minor
* INFO 5 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Return an empty collection instead of null.](https://github.com/Wikia/pandora/blob/6ba50fed4452e1fe7d023bd6e3f6265f62b8d36c/service/discussion/src/main/java/com/wikia/discussionservice/domain/hal/output/ForumThreadHalResponse.java#L40) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1168)
* MAJOR [Return an empty collection instead of null.](https://github.com/Wikia/pandora/blob/6ba50fed4452e1fe7d023bd6e3f6265f62b8d36c/service/discussion/src/main/java/com/wikia/discussionservice/domain/hal/output/PostHalResponse.java#L48) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1168)
* MAJOR [This block of commented-out lines of code should be removed.](https://github.com/Wikia/pandora/blob/6ba50fed4452e1fe7d023bd6e3f6265f62b8d36c/service/discussion/src/main/java/com/wikia/discussionservice/domain/hal/output/PostHalResponse.java#L42) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
* MINOR [Define a constant instead of duplicating this literal "No forum found for site id: %s with forum id: %s" 4 times.](https://github.com/Wikia/pandora/blob/6ba50fed4452e1fe7d023bd6e3f6265f62b8d36c/service/discussion/src/main/java/com/wikia/discussionservice/resources/ForumResource.java#L209) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "Invalid response group" 3 times.](https://github.com/Wikia/pandora/blob/6ba50fed4452e1fe7d023bd6e3f6265f62b8d36c/service/discussion/src/main/java/com/wikia/discussionservice/resources/ForumResource.java#L114) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MAJOR [Either log or rethrow this exception.](https://github.com/Wikia/pandora/blob/6ba50fed4452e1fe7d023bd6e3f6265f62b8d36c/service/discussion/src/main/java/com/wikia/discussionservice/resources/PostResource.java#L260) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)
* INFO [Complete the task associated to this TODO comment.](https://github.com/Wikia/pandora/blob/6ba50fed4452e1fe7d023bd6e3f6265f62b8d36c/service/discussion/src/main/java/com/wikia/discussionservice/resources/PostResource.java#L78) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)
* INFO [Complete the task associated to this TODO comment.](https://github.com/Wikia/pandora/blob/6ba50fed4452e1fe7d023bd6e3f6265f62b8d36c/service/discussion/src/main/java/com/wikia/discussionservice/resources/PostResource.java#L283) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)
* INFO [Complete the task associated to this TODO comment.](https://github.com/Wikia/pandora/blob/6ba50fed4452e1fe7d023bd6e3f6265f62b8d36c/service/discussion/src/main/java/com/wikia/discussionservice/resources/PostResource.java#L112) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)
* MAJOR [Either log or rethrow this exception.](https://github.com/Wikia/pandora/blob/6ba50fed4452e1fe7d023bd6e3f6265f62b8d36c/service/discussion/src/main/java/com/wikia/discussionservice/resources/PostResource.java#L228) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)
* INFO [Complete the task associated to this TODO comment.](https://github.com/Wikia/pandora/blob/6ba50fed4452e1fe7d023bd6e3f6265f62b8d36c/service/discussion/src/main/java/com/wikia/discussionservice/resources/ThreadResource.java#L94) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)
* INFO [Complete the task associated to this TODO comment.](https://github.com/Wikia/pandora/blob/6ba50fed4452e1fe7d023bd6e3f6265f62b8d36c/service/discussion/src/main/java/com/wikia/discussionservice/resources/ThreadResource.java#L285) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)
* MINOR [Move this constructor to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/6ba50fed4452e1fe7d023bd6e3f6265f62b8d36c/service/user-attribute/src/main/java/com/wikia/services/userattribute/resources/AttributeResource.java#L48) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MAJOR [Remove those useless parentheses.](https://github.com/Wikia/pandora/blob/6ba50fed4452e1fe7d023bd6e3f6265f62b8d36c/service/user-preference/src/main/java/com/wikia/userpreference/UserPreferenceApplication.java#L35) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessParenthesesCheck)

**Pull request merged**